Sources/MapboxDirections/Incident.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+/**
+ `Incident` describes any corresponding event, used for annotating the route.
+ */
+public struct Incident: Codable, Equatable {
+
+    private enum CodingKeys: String, CodingKey {
+        case identifier = "id"
+        case type
+        case description = "description"
+        case creationTime = "creation_time"
+        case startTime = "start_time"
+        case endTime = "end_time"
+        case impact = "impact"
+        case subtype = "sub_type"
+        case subtypeDescription = "sub_type_description"
+        case alertCodes = "alertc_codes"
+        case lanesBlocked = "lanes_blocked"
+        case geometryIndexStart = "geometry_index_start"
+        case geometryIndexEnd = "geometry_index_end"
+    }
+
+    var identifier: String
+    var type: String
```

Comment:
What are the valid values of `type` and `subtype`, or is they freeform? If there are sets of valid values, we should make them into enumerations ideally.